core/src/autogluon/core/features/infer_types.py

```diff
@@ -136,3 +136,26 @@ def check_if_nlp_feature(X: Series) -> bool:
         return False
 
     return True
+
+
+def get_bool_true_val(series: pd.Series, replace_val=None):
+    """
+    From a pandas series, get the replace_val to convert to boolean when calling:
+    series_bool = series == replace_val
+
+    series must have exactly 2 unique values
+    """
+    if replace_val is None:
+        try:
+            uniques = series.unique()
```

Comment:
Design decision: so we want to treat `0` and `0.0` values (`int` and `float` types) as the same value?
```python
pd.DataFrame({'a': [0, 0.0]}).a.unique()  # array([0.])
```

Another point, this will auto-cast numeric outputs and `sort()` call won't detect mixed types for numeric types:
```python
pd.DataFrame({'a': [0, 0.0, 1]}).a.unique()  # array([0., 1.]): 1 -> 1.
```
Assuming 0 and 0.0 are treated the same, I would recommend commenting this assumption here

Reply:
Do we handle common conventions anywhere what's considered as `true`? With the current logic series might have confusing interpretations (I understand it's not really important for prediction quality):
```python
['true', 'false']  # True == ' true'
['true', 'zzzz']   # True == ' zzz'
```

Reply:
added comments on this. Yes we want to treat 0 and 0.0 as the same.

For now I'd prefer avoiding hard-coded 'true' values for simplicity. If we want to later on, we can borrow from our code that determines positive class when label is binary.